src/components/content/tags/ListTags.js

```diff
@@ -20,28 +20,62 @@ const Button = styled.button`
     color: #FFFFFF;
     border: 1px solid #491C8A;
     cursor: pointer;
+
+    :hover {
+        color: #491C8A;
+        border: 1px solid  #491C8A;
+        background-color: #FFFFFF;
+    }
 `
 
 const ListContainer = styled.div`
     display: flex;
     justify-content: space-between;
     flex-wrap: wrap;
 
-    @media (max-width: 520px) {
+    @media (max-width: 340px) {
         justify-content: center;
     }
 `
 
-const ListTags = ({ tags, dispatchFilter, selectedArrTags }) => (
-    <ListContainer>
-        <ShowMoreContainer>
-            <Button value = "ShowMore">Show More Tags...</Button>
-        </ShowMoreContainer>
-        {
-            tags.map(tag =>
-                <Tag dispatchFilter={ dispatchFilter } key={ tag } tag={ tag } isSelected={ selectedArrTags.includes(tag) } />)
+
+class ListTags extends React.Component {
+
+    constructor(props){
+        super(props)
+        this.state = {
+            itemsToShow: 8,
+            expanded: false  
         }
-    </ListContainer>
-)
+        this.showMore = this.showMore.bind(this);
+    }
+
+    showMore() {
+        this.state.itemsToShow === 8 ? (
+          this.setState({ itemsToShow: 48, expanded: true })
+        ) : (
+          this.setState({ itemsToShow: 8, expanded: false })
+        )
+    }
+
+    render(){
+        const { tags, dispatchFilter, selectedArrTags } = this.props
+        return<ListContainer>
+            <ShowTags>
+                <Button value = "ShowMore" onClick={this.showMore}>
+                {this.state.expanded ? (
```

Comment:
{ this.state.expanded && <span>Show less tags...</span> }
{ !this.state.expanded && <span>Show more tags...</span> }

Reply:
Indentation